gradle-plugin/src/main/groovy/mozilla/telemetry/glean-gradle-plugin/GleanGradlePlugin.groovy

```diff
@@ -49,42 +49,55 @@ class GleanPlugin implements Plugin<Project> {
 import importlib
 import subprocess
 import sys
-module_name = sys.argv[1]
-expected_version = sys.argv[2]
+offline = sys.argv[1] == "offline"
+module_name = sys.argv[2]
+expected_version = sys.argv[3]
 try:
     module = importlib.import_module(module_name)
 except ImportError:
     found_version = None
 else:
     found_version = getattr(module, '__version__')
 if found_version != expected_version:
-    subprocess.check_call([
-        sys.executable,
-        '-m',
-        'pip',
-        'install',
-        '--upgrade',
-        f'{module_name}=={expected_version}'
-    ])
+    if not offline:
+        subprocess.check_call([
+            sys.executable,
+            '-m',
+            'pip',
+            'install',
+            '--upgrade',
+            f'{module_name}=={expected_version}'
+        ])
+    else:
+        print(f"Using Python environment at {sys.executable},")
+        print(f"expected glean_parser version {expected_version}, found {found_version}.")
+        sys.exit(1)
 try:
     subprocess.check_call([
         sys.executable,
         '-m',
         module_name
-    ] + sys.argv[3:])
+    ] + sys.argv[4:])
 except:
     # We don't need to show a traceback in this helper script.
     # Only the output of the subprocess is interesting.
     sys.exit(1)
 """
 
-    static File getPythonCommand(File condaDir) {
+    // Are we doing an offline build (by passing `--offline` to `./gradle`)?
+    private Boolean isOffline
+
+    static File getPythonCommand(File envDir) {
```

Comment:
```suggestion
    static File getPythonCommand(File envDir, boolean isOffline) {
```

Otherwise Gradle complains about accessing a member from the static function.